### PR TITLE
CSS: Clarify JS relationship with :fullscreen

### LINKED ISSUE #29856: Confusing claim on JS for :fullscreen
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/:fullscreen

### What specific section or headline is this issue about?

Examples

### What information was incorrect, unhelpful, or incomplete?

> In this example, the color of a button is changed depending on whether or not the document is in fullscreen mode. This is done without needing to specifically apply style changes using JavaScript.

In a web development support group on Matrix, a beginner was confused by the last sentence.
The example showed a demo with a button, but no event handler on the button to [toggle fullscreen mode](https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API/Guide#activating_fullscreen_mode).

As such the example appears to be incomplete.

### What did you expect to see?

I could imagine that either JS is added to request fullscreen mode (as hinted in other places on the subject) or that F11 is an OS-supported way to enter the fullscreen mode or that a sentence is added to explain, that the example is valid once the fullscreen mode is entered.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/_colon_fullscreen`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/:fullscreen
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_colon_fullscreen/index.md
* Last commit: https://github.com/mdn/content/commit/6b1e3eebf22abf1b73bb219581335b1147b75d7a
* Document last modified: 2023-03-12T09:29:44.000Z

</details>

### PR DESCRIPTION
**Pull request by Ryuno-Ki**
### Description

Explain the relationship on how to enter Fullscreen so that the example can be replicated.

### Motivation

In a support channel for web development, a beginner asked for help with this section. We were successful in clarifying the confusion.

### Related issues and pull requests

Fixes #29856.

### CODE REVIEW

File: files/en-us/web/css/_colon_fullscreen/index.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ The `:fullscreen` pseudo-class lets you configure your stylesheets to automatica
 
 ## Examples
 
-In this example, the color of a button is changed depending on whether or not the document is in fullscreen mode. This is done without needing to specifically apply style changes using JavaScript.
+In this example, the color of a button is changed depending on whether or not the document is in fullscreen mode. You will need JavaScript to [request fullscreen mode](/en-US/docs/Web/API/Fullscreen_API/Guide). Once in that mode, no JavaScript is needed to style.
```
Comment (estelle) [inline files/en-us/web/css/_colon_fullscreen/index.md:26]:
```suggestion
In this example, the color of a button is changed depending on whether or not the document is in fullscreen mode. While JavaScript is needed to request [fullscreen mode](/en-US/docs/Web/API/Fullscreen_API/Guide), the selector and color change are CSS features.
```

The original makes it sound like you need JS to use this feature. This suggested edit downplays that.

**Comment by OnkarRuikar** [comment]
There is a bug in the code. The `:fullscreen` works only on the element which has fullscreen flag set to true. It doesn't apply to child elements. `#fs-toggle:fullscreen` doesn't work if parent div is taken full screen and no one would ever take only button fullscreen. Ideally the selectors need to be like `#parent:fullscreen > #fs-toggle-button`.

To really "complete" the example we can provide the entire code https://jsfiddle.net/k0984wxy/ .
You can use [LiveSample macro](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Live_samples) to make the demo interactive.

**Comment by OnkarRuikar** [comment]
The page intro also seems wrong:
> The **`:fullscreen`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches every element which is currently in fullscreen mode.

I tested this on latest Chrome and Firefox. It doesn't match every element in fullscreen mode. `:fullscreen {}` is not same as `*{}` in fullscreen mode. 

>  If multiple elements have been put into fullscreen mode, this selects them all.

We can't put multiple elements in fullscreen mode because [`Document.fullscreenElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenElement) property can point to only one element at a time. When you put a div in fullsreen its children do not get full screen flag set and the `:fullscreen` selector doesn't select them.

**Comment by yookoala** [comment]
The change that this PR addressed is probably covered by the changes in #29951.

**Comment by Ryuno-Ki** [comment]
Closing in favour of #29951 

See associated issue for reasoning.

**Pull request closed without merging**